Source/HTTP/WinHttp/winhttp_provider.cpp

```diff
@@ -342,31 +358,40 @@ Result<HINTERNET> WinHttpProvider::GetHSession(uint32_t securityProtocolFlags)
     m_proxyType = get_ie_proxy_info(proxy_protocol::https, proxyUri);
     GetProxyName(m_proxyType, proxyUri, accessType, wProxyName);
 
+    // Determine WinHTTP flags based on URL scheme
+    // Use WINHTTP_FLAG_SECURE_DEFAULTS for HTTPS and WINHTTP_FLAG_ASYNC for HTTP
+    DWORD openFlags;
+    if (isHttps)
+    {
+        // For HTTPS, use secure defaults which implies WINHTTP_FLAG_ASYNC
+        openFlags = WINHTTP_FLAG_SECURE_DEFAULTS;
+    }
+    else
+    {
+        // For HTTP, use async only (allow insecure connections)
+        openFlags = WINHTTP_FLAG_ASYNC;
```

Comment:
If I understood correctly, HTTP being broken was something you found incidentally during your own out-of-band testing. Should you add automated tests to make sure HTTP continues to work for all HTTP-supporting platforms?

Reply:
fixed, test added